### PR TITLE
Filter functions

### PR DESCRIPTION
**Pull request by artemkovalyov**
I added a bit info about filter functions. Please, take a look and give suggestions for corrections. This was asked by a customer in today's feedback session.

### CODE REVIEW

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 1]
```diff
@@ -330,14 +330,31 @@ To achieve a different order with `and` as the top level statement one would nes
 
 <!-- TODO: Explain filters on complex types and navigational properties -->
 
-The [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/) allows for a limited range of filter expressions compared to OData v4. A detailed list of what is available in the SDK can be obtained from [the Javadoc](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/helper/package-summary.html). The functionality can also be discovered through the fluent API.
+Please, check what filter options are supported by [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/). Search for: **4.5. Filter System Query Option ($filter)** after navigation to the link.
```
Comment (Emdee89) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:333]:
```suggestion
We recommend to check what filter options are supported by [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/). Search for: **4.5. Filter System Query Option ($filter)** after navigation to the link.
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 2]
```diff
@@ -330,14 +330,31 @@ To achieve a different order with `and` as the top level statement one would nes
...
+| Less than             | `lt`            | /Products?$filter=Price lt 20                |
+| Greater than          | `gt`            | /Products?$filter=Price gt 20                |
+| Greater than or equal | `ge`            | /Products?$filter=Price ge 10                |
+| Less than or equal    | `le`            | /Products?$filter=Price le 100               |
+
+:::note  "Header"
+We are working on supporting string filtering function like `SubstringOf`, `endsWith`, `startsWith`, etc.
+:::
+
+For more details check our [API documentation](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/helper/package-summary.html) or use `autocomplete` function of your IDE to see available methods of the provided Fluent API.
```
Comment (Emdee89) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:350]: I would directly refer to the Javadoc of the class "EntityField" (https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/helper/EntityField.html) as it contains the list of supported filter functions.

Comment (MatKuhr) [reply docs/java/features/odata/type-safe-client-odata-v2.mdx:350]: Agreed, and I would do the same for OData V4.

Comment (MatKuhr) [reply docs/java/features/odata/type-safe-client-odata-v2.mdx:350]: e.g. https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odatav4/expression/FilterableValue.html

**Comment by artemkovalyov** [comment]
Thanks for the review @Emdee89  and @MatKuhr, I'll give it a thought and get back to it later this week.

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 3]
```diff
@@ -330,14 +330,31 @@ To achieve a different order with `and` as the top level statement one would nes
 
 <!-- TODO: Explain filters on complex types and navigational properties -->
 
-The [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/) allows for a limited range of filter expressions compared to OData v4. A detailed list of what is available in the SDK can be obtained from [the Javadoc](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/helper/package-summary.html). The functionality can also be discovered through the fluent API.
+We recommend to check what filter options are supported by [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/). Search for: **4.5. Filter System Query Option ($filter)** after navigation to the link.
```
Comment (florian-richter) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:333]:
```suggestion
We recommend to check what filter options are supported by [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/uri-conventions/). 
Search for: **4.5. Filter System Query Option ($filter)** after navigation to the link.
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 4]
```diff
@@ -330,14 +330,31 @@ To achieve a different order with `and` as the top level statement one would nes
...
+|-----------------------|-----------------|----------------------------------------------|
+| Equals                | `eq`            | /Suppliers?$filter=Address/City eq 'Redmond' |
+| No Equals             | `ne`            | /Suppliers?$filter=Address/City ne 'London'  |
+| Less than             | `lt`            | /Products?$filter=Price lt 20                |
+| Greater than          | `gt`            | /Products?$filter=Price gt 20                |
+| Greater than or equal | `ge`            | /Products?$filter=Price ge 10                |
+| Less than or equal    | `le`            | /Products?$filter=Price le 100               |
+
+:::note  "Header"
+We are working on supporting string filtering function like `SubstringOf`, `endsWith`, `startsWith`, etc.
```
Comment (florian-richter) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:347]:
```suggestion
We are working on supporting string filtering function like `substringOf`, `endsWith`, `startsWith`, etc.
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 5]
```diff
@@ -330,14 +330,31 @@ To achieve a different order with `and` as the top level statement one would nes
...
+For more details check our [API documentation](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/helper/package-summary.html) or use `autocomplete` function of your IDE to see available methods of the provided Fluent API.
 
 <!--
 ### Count ###
 -->
 
-### Batch Requests 
-Batch requests allow wrapping multiple OData requests into one single batch call. Thereby we reduce the number of roundtrips to the remote server. Refer to the [official OData V2 spec](https://www.odata.org/documentation/odata-version-2-0/batch-processing/) 
+### Batch Requests
+Batch requests allow wrapping multiple OData requests into one single batch call. Thereby we reduce the number of roundtrips to the remote server. Refer to the [official OData V2 spec](https://www.odata.org/documentation/odata-version-2-0/batch-processing/)
```
Comment (florian-richter) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:357]:
```suggestion
Batch requests allow wrapping multiple OData requests into one single batch call to reduce the number of roundtrips to the remote server. 
Refer to the [OData v2 standard](https://www.odata.org/documentation/odata-version-2-0/batch-processing/).
```

**Review by florian-richter — COMMENTED** [review]
Some formatting recommendations:

File: docs/java/features/odata/type-safe-client-odata-v2.mdx (hunk 4 above)
Comment (MatKuhr) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:347]: Support for this is already implemented

**Review by MatKuhr — CHANGES_REQUESTED** [review]
I am not sure how meaningful it is to document the filter functions here. We are mainly copy/pasting the OData docs. Also we would have to repeat it again for V4.

I would rather create a note for any operations we don't yet support. To my knowledge that would be e.g. `in` for OData V4.

As far as I have seen the main source for people missing the functionality is if they are on an older version where it's not included.

**Pull request closed without merging**